cqf-fhir-cr/src/main/java/org/opencds/cqf/fhir/cr/measure/dstu3/Dstu3MeasureDefBuilder.java

```diff
@@ -167,25 +168,51 @@ private MeasureScoring getMeasureScoring(Measure measure) {
     }
 
     private boolean isIncreaseImprovementNotation(String improvementNotationValue) {
+        validateImprovementNotationCode(improvementNotationValue);
         return improvementNotationValue.equals(IMPROVEMENT_NOTATION_SYSTEM_INCREASE);
     }
 
-    public boolean measureIsIncreaseImprovementNotation(Measure measure) {
-        if (measure.hasImprovementNotation()) {
-            return isIncreaseImprovementNotation(measure.getImprovementNotation());
-        } else {
-            // default ImprovementNotation behavior
-            return true;
+    private void validateImprovementNotationCode(String improvementNotationValue) {
+        boolean hasValidCode = IMPROVEMENT_NOTATION_SYSTEM_INCREASE.equals(improvementNotationValue)
+                || IMPROVEMENT_NOTATION_SYSTEM_DECREASE.equals(improvementNotationValue);
+        if (!hasValidCode) {
+            throw new IllegalArgumentException(String.format(
+                    "ImprovementNotation Coding has invalid code: %s, combination for Measure.",
+                    improvementNotationValue));
+        }
+    }
+
+    public String getGroupImprovementNotationExt(Measure.MeasureGroupComponent group) {
+        StringType value = (StringType) group.getExtensionByUrl(MEASUREREPORT_IMPROVEMENT_NOTATION_EXTENSION)
```

Comment:
Wll this always be a StringType?  Would it be a good idea to do an instanceof check here first?